mw-webapp/src/service/WayService.ts

```diff
@@ -64,18 +68,43 @@ export type GetWaysFilter = {
  */
 export class WayService {
 
+  /**
+   * Get WaysDTO amount
+   */
+  public static async getWaysDTOAmount(): Promise<number> {
+    const waysRef = collection(db, PATH_TO_WAYS_COLLECTION);
+    const snapshot = await getCountFromServer(waysRef);
+    const waysAmount = snapshot.data().count;
+
+    const readsAmount = Math.ceil(waysAmount / AMOUNT_DOCS_FOR_COUNT_READS);
+
+    logToConsole(`WayService:getWaysDTOAmount: ${readsAmount} ${RequestOperations.READ} operations`);
+
+    return waysAmount;
+  }
+
   /**
    * Get WaysDTO
    */
-  public static async getWaysDTO(): Promise<WayDTO[]> {
+  public static async getWaysDTO(lastWayUuid?: string): Promise<WayDTO[]> {
     const waysRef = collection(db, PATH_TO_WAYS_COLLECTION);
-    const waysOrderedByName = query(waysRef, orderBy(WAY_CREATED_AT_FIELD, "desc"));
+
+    /**
+     * ExtraRequest
+     * TODO: need to find how to convert object to DocumentSnapshot<DocumentData, DocumentData>
```

Comment:
Why do we need to find this? Where the issued number?

Reply:
Deleted